### ISSUE TITLE
Linux HW decode NOT work

### ISSUE THREAD

**Issue by wx5391805** [StaZhu/enable-chromium-hevc-hardware-decoding#48]
Linux arm64 fails, `vaapi_video_decoder.cc` should now only serve ChromeOS. `vaapi_video_video_decode_accelerator.cc` is the Linux path now. Have you tried it on Linux?  Chromium 110 is somewhat different from the previous version on Linux hardware acceleration. Thank you very much.

**Comment by wx5391805** [comment]
Well, I have to correct it, it’s really bad under Linux, there are two decoders, `VdaVideoDecoder` and `VaapiVideoDecoder`, HEVC code is in the latter, the choice of decoder is limited by several switches such as UseChromeOSDirectVideoDecoder, Vulkan, etc., and it has something to do with the version.

**Comment by StaZhu** [comment]
I think chromium media has a plan to completely remove the use of `VDAVideoDecoder` and finally switch to `VaapiVideoDecoder`, however since vaapi isn’t supported well on intel iGPU devices and not to mention that amd and nvidia gpus doesnt officially support vaapi at all, and Linux arm64 devices dont use vaapi as well. So i think i can’t help to solve this for now since the infrastructure on linux chrome is really bad and the ROI is too low.

**Comment by wx5391805** [comment]
Yeah I agree with you. The fact is that I do use vaapi on my Linux arm64 now. It works on Chromium 102. It's just that linux users with NV GPUs can't use `VaapiVideoDecoder` because of the drive. Unfortunately, `VdaVideoDecoder` also doesn't work since 110, which prevents me from upgrading.

**Comment by StaZhu** [comment]
> NVIDIA and AMD have now focused on VulkanVideo, unfortunately the Chrome Media team has no intention of implementing Vulkan decoding. But the situation may change in the future, as many Android OEM/IHV seem to be very interested in VulkanVideo

Can you fire a new crbug issue and ask chromium folks if they have a plan to implement vulkan video support? The decision completely depends on chrome media team i think.

**Comment by StaZhu** [comment]
> I downloaded the deb and finally i can watch hevc in a browser thanks. My only question is how can i set this up to work in brave which is my main browser?

You can’t, to integrate SW decoding, browser needs to cherry pick the SW commit, and I don’t think brave want to do that.
